pkg/machines/components/create-vm-dialog/createVmDialogUtils.js

```diff
@@ -35,6 +37,23 @@ const IGNORE_VENDORS = ['ALTLinux', 'Mandriva', 'GNOME Project'];
 const ACCEPT_RELEASE_DATES_AFTER = getTodayYearShifted(-3);
 const ACCEPT_EOL_DATES_AFTER = getTodayYearShifted(-1);
 
+/*
+ * Uses libosinfo to autodetect an OS and distro from a URL - This currently works only for rpm based distros
+ * @param {string} url - A URL pointing to the media or the tree.
+ */
+export function autodetectOS(url) {
+    let urlType = 'tree';
+
+    if (url.endsWith('.iso'))
+        urlType = 'media';
+
+    // osinfo-detect uses GIO to read the tree info file over http. cockpit-bridge used to unset GIO env variables
+    // which blocked us from using GIO calls over cockpit-bridge.
+    // Overwrite the env vars here, untill commit https://github.com/cockpit-project/cockpit/pull/11567/commits/6c8c79b54e8d8164e540267299e1bf4cb3184c7f
+    // exists in all supported distros.
+    return cockpit.spawn(['osinfo-detect', '--type', urlType, url], { environ: ['GIO_USE_VFS=gvfs'], err: 'message' });
```

Comment:
Please let's add `LC_ALL=en_US.UTF-8` to ensure that this doesn't fall into a translation trap. (This is why `env` format would be really preferrable, as it's machine readable)

Reply:
This should handle failures either here or in createVmDialog, to avoid an unhandled rejection error in the JS console. IMHO handling it here with a console.log is fine.

Reply:
Done